src/PostgREST/Middleware.hs

```diff
@@ -64,12 +68,12 @@ runPgLocals conf claims app req jsonDbS = do
   where
     methodSql = setConfigLocal mempty ("request.method", iMethod req)
     pathSql = setConfigLocal mempty ("request.path", iPath req)
-    headersSql = setConfigLocal "request.header." <$> iHeaders req
-    cookiesSql = setConfigLocal "request.cookie." <$> iCookies req
+    headersSql = setGuc "request.header." "request.headers" (iHeaders req)
+    cookiesSql = setGuc "request.cookie." "request.cookies" (iCookies req)
     claimsWithRole =
       let anon = JSON.String . toS $ configDbAnonRole conf in -- role claim defaults to anon if not specified in jwt
       M.union claims (M.singleton "role" anon)
-    claimsSql = setConfigLocal "request.jwt.claim." <$> [(toS c, toS $ unquoted v) | (c,v) <- M.toList claimsWithRole]
+    claimsSql = setGuc "request.jwt.claim." "request.jwt.claims" [(toS c, toS $ unquoted v) | (c,v) <- M.toList claimsWithRole]
```

Comment:
I think this could be optimized. Here `M.toList claimsWithRole` it looks like it's doing a conversion from HashMap to List, later the `gucJsonVal` function will do the opposite. Maybe avoid this by encoding the `claimsWithRole` directly.